python/dgl/mock_sparse2/sparse_matrix.py

```diff
@@ -0,0 +1,362 @@
+        """Get the coordinate (COO) representation of the sparse matrix.
+
+        Returns
+        -------
+        Tuple[torch.Tensor, torch.Tensor, torch.Tensor]
+            A tuple of tensors containing row, column coordinates and values.
+        """
+        return self.c_sparse_matrix.csr()
+
+    def csc(self) -> Tuple[torch.Tensor, ...]:
+        """Get the coordinate (COO) representation of the sparse matrix.
+
+        Returns
+        -------
+        Tuple[torch.Tensor, torch.Tensor, torch.Tensor]
+            A tuple of tensors containing row, column coordinates and values.
+        """
+        return self.c_sparse_matrix.csc()
+
+
+def create_from_coo(
+    row: torch.Tensor,
+    col: torch.Tensor,
+    val: Optional[torch.Tensor] = None,
+    shape: Optional[Tuple[int, int]] = None,
+) -> SparseMatrix:
+    """Create a sparse matrix from row and column coordinates.
+
+    Parameters
+    ----------
+    row : tensor
+        The row indices of shape (nnz).
+    col : tensor
+        The column indices of shape (nnz).
+    val : tensor, optional
+        The values of shape (nnz) or (nnz, D). If None, it will be a tensor of shape (nnz)
+        filled by 1.
+    shape : tuple[int, int], optional
+        If not specified, it will be inferred from :attr:`row` and :attr:`col`, i.e.,
+        (row.max() + 1, col.max() + 1). Otherwise, :attr:`shape` should be no smaller
+        than this.
+
+    Returns
+    -------
+    SparseMatrix
+        Sparse matrix
+
+    Examples
+    --------
+
+    Case1: Sparse matrix with row and column indices without values.
+
+    >>> src = torch.tensor([1, 1, 2])
+    >>> dst = torch.tensor([2, 4, 3])
+    >>> A = create_from_coo(src, dst)
+    >>> print(A)
+    SparseMatrix(indices=tensor([[1, 1, 2],
+                                 [2, 4, 3]]),
+                 values=tensor([1., 1., 1.]),
+                 shape=(3, 5), nnz=3)
+    >>> # Specify shape
+    >>> A = create_from_coo(src, dst, shape=(5, 5))
+    >>> print(A)
+    SparseMatrix(indices=tensor([[1, 1, 2],
+                                 [2, 4, 3]]),
+                 values=tensor([1., 1., 1.]),
+                 shape=(5, 5), nnz=3)
+
+    Case2: Sparse matrix with scalar/vector values. Following example is with
+    vector data.
+
+    >>> val = torch.tensor([[1, 1], [2, 2], [3, 3]])
+    >>> A = create_from_coo(src, dst, val)
+    SparseMatrix(indices=tensor([[1, 1, 2],
+                                 [2, 4, 3]]),
+                 values=tensor([[1, 1],
+                                [2, 2],
+                                [3, 3]]),
+                 shape=(3, 5), nnz=3)
+    """
+    if shape is None:
+        shape = (torch.max(row).item() + 1, torch.max(col).item() + 1)
+    if val is None:
+        val = torch.ones(row.shape[0])
+
+    return SparseMatrix(
+        torch.ops.dgl_sparse.create_from_coo(row, col, val, shape)
+    )
+
+
+# FIXME: The docstring cannot print A because we cannot print
```

Comment:
Remove this `FIXME` as it's no longer true.

Reply:
The examples in the CSR/CSC docstring cannot run because in the `__repr__` we need to convert the CSR/CSC formats to COO.